File: modules/base/test/matlab/TrialTest.m
classdef TrialTest < matlab.unittest.TestCase
    methods (Test)
    
        function events(testCase)
            t = ma.Trial('trial');
            e = t.events();
            testCase.verifyEqual(t.hasChildren(), true);
            testCase.verifyEqual(e.hasParents(), true);
            testCase.verifyEqual(t.name(), 'trial');
            testCase.verifyEqual(e.name(), 'Events');
            testCase.verifyEqual(double(t.refcount()), 1);
            testCase.verifyEqual(double(e.refcount()), 2);
            delete(t);
            testCase.verifyEqual(e.hasParents(), false);
            testCase.verifyEqual(double(e.refcount()), 1);
        end
        
        function timesequences(testCase)
            t = ma.Trial('trial');
            ts = t.timeSequences();
            testCase.verifyEqual(t.hasChildren(), true);
            testCase.verifyEqual(ts.hasParents(), true);
            testCase.verifyEqual(t.name(), 'trial');
            testCase.verifyEqual(ts.name(), 'TimeSequences');
            testCase.verifyEqual(double(t.refcount()), 1);
            testCase.verifyEqual(double(ts.refcount()), 2);
            delete(t);
            testCase.verifyEqual(ts.hasParents(), false);
            testCase.verifyEqual(double(ts.refcount()), 1);
        end
        
        function hardwares(testCase)
            t = ma.Trial('trial');
            hws = t.hardwares();
            testCase.verifyEqual(t.hasChildren(), true);
            testCase.verifyEqual(hws.hasParents(), true);
            testCase.verifyEqual(t.name(), 'trial');
            testCase.verifyEqual(hws.name(), 'Hardwares');
            testCase.verifyEqual(double(t.refcount()), 1);
            testCase.verifyEqual(double(hws.refcount()), 2);
            delete(t);
            testCase.verifyEqual(hws.hasParents(), false);
            testCase.verifyEqual(double(hws.refcount()), 1);
        end
        
    end
end

File: modules/base/test/python/trialTest.py
import ma
import unittest

class TrialTest(unittest.TestCase):
    
    def test_events(self):
        t = ma.Trial('trial')
        evts = t.events()
        self.assertEqual(t.hasChildren(), True)
        self.assertEqual(evts.hasParents(), True)
        self.assertEqual(t.name(), 'trial')
        self.assertEqual(evts.name(), 'Events')
        self.assertEqual(t.refcount(), 1)
        self.assertEqual(evts.refcount(), 2)
        del t
        self.assertEqual(evts.hasParents(), False)
        self.assertEqual(evts.refcount(), 1)
    
    def test_timesequences(self):
        t = ma.Trial('trial')
        tss = t.timeSequences()
        self.assertEqual(t.hasChildren(), True)
        self.assertEqual(tss.hasParents(), True)
        self.assertEqual(t.name(), 'trial')
        self.assertEqual(tss.name(), 'TimeSequences')
        self.assertEqual(t.refcount(), 1)
        self.assertEqual(tss.refcount(), 2)
        del t
        self.assertEqual(tss.hasParents(), False)
        self.assertEqual(tss.refcount(), 1)
        
    def test_hardwares(self):
        t = ma.Trial('trial')
        hws = t.hardwares()
        self.assertEqual(t.hasChildren(), True)
        self.assertEqual(hws.hasParents(), True)
        self.assertEqual(t.name(), 'trial')
        self.assertEqual(hws.name(), 'Hardwares')
        self.assertEqual(t.refcount(), 1)
        self.assertEqual(hws.refcount(), 2)
        del t
        self.assertEqual(hws.hasParents(), False)
        self.assertEqual(hws.refcount(), 1)
